monitor.cpp

```diff
@@ -1,38 +1,97 @@
-#include "./monitor.h"
-#include <assert.h>
-#include <thread>
-#include <chrono>
+#include "monitor.h"
 #include <iostream>
-using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
 
-int vitalsOk(float temperature, float pulseRate, float spo2) {
-  if (temperature > 102 || temperature < 95) {
-    cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
+std::string language = "EN";
+
+const float TEMP_WARNING_TOLERANCE = 1.53;
+const float PULSE_WARNING_TOLERANCE = 1.5;
+const float SPO2_WARNING_TOLERANCE = 1.35;
+
+bool isTemperatureCritical(float temperature) {
+    return (temperature < 36.0 || temperature > 37.5);
+}
+
+bool isTemperatureWarning(float temperature) {
+    return (temperature >= 36.0 && temperature < 36.0 + TEMP_WARNING_TOLERANCE) ||
+           (temperature > 37.5 - TEMP_WARNING_TOLERANCE && temperature <= 37.5);
+}
+
+bool isPulseRateOutOfRange(float pulseRate) {
+    return (pulseRate < 60 || pulseRate > 100);
+}
+
+bool isPulseRateWarning(float pulseRate) {
+    return (pulseRate >= 60 && pulseRate < 60 + PULSE_WARNING_TOLERANCE) ||
+           (pulseRate > 100 - PULSE_WARNING_TOLERANCE && pulseRate <= 100);
+}
+
+bool isSpo2Low(float spo2) {
+    return spo2 < 90;
+}
+
+bool isSpo2Warning(float spo2) {
+    return (spo2 >= 90 && spo2 < 90 + SPO2_WARNING_TOLERANCE);
+}
+
+const char* getMessage(const char* key) {
+    if (language == "DE") {
```

Comment:
Try a data-structure like std::map to map the key to the message, instead of if-statements

Reply:
Thanks Sudeep , I've made the changes.